pymeasure/instruments/eurotest/eurotestHPP120256.py

```diff
@@ -0,0 +1,320 @@
+         voltage_setting will be present on the output""",
+        validator=strict_discrete_set,
+        values={'ON': 'ON', 'OFF': 'OFF', 'ENable': 'ON', 'DISable': 'OFF',
+                'EN': 'ON', 'DIS': 'OFF'},
+        map_values=True
+    )
+
+    @property
+    def id(self):
+        """ Returns the identification of the instrument """
+        log.info("Requesting instrument identification...")
+
+        response = self.ask("ID")
+        return response.encode(self.response_encoding).decode('utf-8', 'ignore')
+
+    def status(self):
+        """ Returns the unit Status which is a 16bits response where
+        every bit indicates teh state of one subsystem of the HV Source."""
+        log.info("Requesting instrument status...")
+
+        response = self.ask("STATUS,DI")
+        response = response.encode(self.response_encoding).decode('utf-8', 'ignore')
+        log.debug(f"Instrument status = {response}")
+        return response
+
+    def lam_status(self):
+        """ Returns the LAM status which is the status of the untit from the point
+        of view of the process. Fo example, as a response of asking STATUS,LAM, the HV
+        voltage could response one of the messages from the next list:
+        LAM,ERROR External Inhibit occurred during Kill enable
+        LAM,INHIBIT External Inhibit occurred
+        LAM,TRIP ERROR Software current trip occurred
+        LAM,INPUT ERROR Wrong command received
+        LAM,OK Status OK"""
+        log.info("Requesting instrument STATUS LAM...")
+
+        response = self.ask("STATUS,LAM")
+        response = response.encode(self.response_encoding).decode('utf-8', 'ignore')
+        log.debug(f"Instrument STATUS LAM = {response}")
+        return response
+
+    def emergency_off(self):
+        """ The output of the HV source will be switched OFF permanently and the values
+        of the voltage a current settings set to zero"""
+        log.info("Sending emergency off command to the instrument.")
+
+        self.write("EMCY OFF")
+        time.sleep(self.COMMAND_DELAY)
+
+    def shutdown(self, ramp):
+        """
+        Ramps the HV source to zero with a determinated ramp and without waiting
+        to the output reaches zero volts dissables it.
+        :param ramp: indicates the ramp
+        """
+        log.info(f"Executing the shutdown function with ramp: {ramp} V/s.")
+
+        self.ramp_to_zero(ramp)
+        super().shutdown()
+
+    def ramp_to_zero(self, ramp):
+        """
+        Ramps the HV source to zero with a determinated ramp
+        :param ramp: indicates the ramp we want
+        """
+        log.info("Executing the ramp_to_zero function with ramp: {ramp} V/s.")
+
+        self.voltage_ramp = ramp
+        time.sleep(self.COMMAND_DELAY)
+        self.voltage = 0
+        time.sleep(self.COMMAND_DELAY)
+        self.voltage = 0
+        time.sleep(self.COMMAND_DELAY)
+
+    def wait_for_voltage_output_set(self, check_period, timeout):
```

Comment:
I recommend to add some sensible default values for check_period and timeout. That way a user may just call wait_for.. without supplying further values, unless needed.

Reply:
Done. Thanks